zebra/zebra_vty.c

```diff
@@ -780,10 +784,10 @@ static void show_nexthop_json_helper(json_object *json_nexthop,
 		break;
 	}
 
-	if ((nexthop->vrf_id != re->vrf_id)
-	    && (nexthop->type != NEXTHOP_TYPE_BLACKHOLE))
+	if (nexthop->vrf_id != re->vrf_id) {
```

Comment:
I'm curious, why this can't be always added (JSON vrf)?

Reply:
Whenever we import a route or if nexthop is in another vrf,
in both scenarios above condition will be true. 
In the normal cases i.e., for local routes, they are already
displayed under specific vrf section.
If we add it always then it becomes confusing to
distinguish between local routes that are being redistributed and
imported routes (or even routes with nexthop in different vrf).